src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -403,15 +405,26 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				this._connectionDialog.databaseDropdownExpanded = this.uiController.databaseDropdownExpanded;
 				this.handleOnCancel(this._connectionDialog.newConnectionParams);
 			});
-			this._connectionDialog.onConnect((profile) => this.handleOnConnect(this._connectionDialog.newConnectionParams, profile));
+			this._connectionDialog.onConnect((profile) => {
+				this.handleOnConnect(this._connectionDialog.newConnectionParams, profile);
+			});
 			this._connectionDialog.onShowUiComponent((input) => this.handleShowUiComponent(input));
 			this._connectionDialog.onInitDialog(() => this.handleInitDialog());
 			this._connectionDialog.onFillinConnectionInputs((input) => this.handleFillInConnectionInputs(input));
 			this._connectionDialog.onResetConnection(() => this.handleProviderOnResetConnection());
 			this._connectionDialog.render();
+			this._previousProviderType = this._currentProviderType;
 		}
 		this._connectionDialog.newConnectionParams = params;
-
+		// if provider changed
+		if ((this._previousProviderType !== this._currentProviderType) ||
+			// or if currentProvider not set correctly yet
+			!(this._currentProviderType === Constants.cmsProviderDisplayName && params.providers && params.providers.length > 1)) {
+			this._previousProviderType = undefined;
+			this._connectionDialog.updateProvider(this._providerNameToDisplayNameMap[this.getDefaultProviderName()]);
```

Comment:
Bleh, yeah this really isn't nice. As we discussed before I'm ok with checking it in as-is but you should immediately go back and look at removing these kind of checks otherwise it'll be that much more painful later on.